builtin/aws/ecs/platform.go

```diff
@@ -910,6 +918,78 @@ func (p *Platform) resourceServiceDestroy(
 	return nil
 }
 
+func (p *Platform) resourceAlbRedirectListenerCreate(
+	ctx context.Context,
+	sg terminal.StepGroup,
+	sess *session.Session,
+	log hclog.Logger,
+	alb *Resource_Alb,
+	externalIngressPort ExternalIngressPort,
+
+	state *Resource_Alb_RedirectListener,
+) error {
+
+	s := sg.Add("Initiating ALB Redirect Listener creation")
+	defer s.Abort()
+
+	var HTTPListener *elbv2.Listener
+
+	if externalIngressPort == 443 {
+		elbsrv := elbv2.New(sess)
+
+		listeners, err := elbsrv.DescribeListenersWithContext(ctx, &elbv2.DescribeListenersInput{
+			LoadBalancerArn: &alb.Arn,
+		})
+
+		if err != nil {
+			return status.Errorf(codes.Internal, "failed to create redirect listener: %s", err)
+		}
+
+		for _, listener := range listeners.Listeners {
+			if *listener.Protocol == "HTTP" {
+				HTTPListener = listener
+			}
+		}
+
+		if HTTPListener != nil && *HTTPListener.DefaultActions[0].Type == "forward" {
+			s.Update("HTTP listener use for forwarding. Please destroy previus deployment or try again deploy app for create redirect listener")
+		} else {
```

Comment:
Currently, it looks like if we already created a redirect listener, we'll try to create it again. I don't think we should ever try creating a listener if it already exists. 

We should also check to see if the settings of the existing listener match what we would create, and give some nice output in that case, like "Using existing redirect ALB listener".